src/main/java/com/iota/iri/network/Node.java

```diff
@@ -251,8 +252,9 @@ public void preProcessReceivedData(byte[] receivedData, SocketAddress senderAddr
 
                     if (!cached) {
                         //if not, then validate
-                        receivedTransactionViewModel = TransactionValidator.validate(receivedData, transactionValidator.getMinWeightMagnitude());
+                        receivedTransactionViewModel = new TransactionViewModel(receivedData, Hash.calculate(receivedData, TransactionViewModel.TRINARY_SIZE, SpongeFactory.create(SpongeFactory.Mode.CURLP81)));
```

Comment:
We shouldn't create a new instance each run here.

Reply:
I think it's fine (better, even)

Reply:
I agree with Paul, unless you can show that creating instances causes an issue.